client/src/client-only-routes/show-certification.tsx

```diff
@@ -314,6 +315,10 @@ const ShowCertification = (props: ShowCertificationProps): JSX.Element => {
     </Row>
   );
 
+  if (user.isCheater || user.isBanned) {
+    return <Profile user={user} isSessionUser={false}></Profile>;
+  }
```

Comment:
This is probably too nitpicky, but it is something I have been learning/considering a lot lately:
- If we move this block of code further up, we can reduce the amount of unneeded computation as well as memory used.

---

Otherwise, I also think we should show different messages, depending on whether the user is `isBanned` or `isCheater`. One of the main reasons for having the ability to have the two flags is to give users some reason for their rejection.

---

I am unsure about the `client`/`server` validation. For the most part, we are void of liability, if we hide inappropriate content; if someone goes above and beyond to try access/view the content, then that is on them.

That being said, obviously, it is best to have serverside checks.

Reply:
> One of the main reasons for having the ability to have the two flags is to give users some reason for their rejection.

So I've been thinking about this, and my understanding was the initial copy "ineligible to claim certs" is intentionally vague. That is, we don't want to be accusing someone of violating any policy - that happens privately if they reach out to support.

Reply:
I would believe so. However, unless I misunderstood, Quincy specifically said we wanted users to know (to a degree) why they are flagged.